lib/AST/Module.cpp

```diff
@@ -3304,13 +3304,23 @@ void SourceFile::lookupImportedSPIGroups(
   }
 }
 
+bool shouldImplicitImportAsSPI(ArrayRef<Identifier> spiGroups) {
+  for (auto group : spiGroups) {
+    if (group.str() == "_")
```

Comment:
Typically, `_` in source code is represented in the AST as an `Identifier()` and tested for with `group.empty()`. There are matching quirks in the places you’re parsing and interpreting SPI groups that cause them to be represented as actual identifiers containing “_", but I’d recommend doing the standard thing here.